File: experiments/006-SmalltalkVsJava/cells/java-gradle_1-Empty_free/2022-1c-recuperatorio-parcial1/20260915-110924-39025/workdir/src/main/java/customerimporter/DNI.java
package customerimporter;

import java.util.function.Supplier;

public class DNI extends Identification {

    public static final String TYPE = "D";

    // error messages

    public static String invalidDNINumberErrorDescription() {
        return "Invalid DNI number";
    }

    // initialization

    DNI(String aNumber) {
        super(aNumber);
    }

    // identification

    public String type() {
        return TYPE;
    }

    // type

    public boolean isDNI() {
        return true;
    }

    public Object dniNumberIfNone(Supplier<Object> aNoneBlock) {
        return dniNumber();
    }

    public int dniNumber() {
        return Integer.parseInt(number());
    }

    // validation

    protected void assertIsValid() {
        assertThat(Characters.areAllDigits(number()));
        assertThat(dniNumber() >= 1 && dniNumber() <= 99999999);
    }

    protected String invalidNumberErrorDescription() {
        return invalidDNINumberErrorDescription();
    }
}

File: experiments/020-PythonVsJava/cells/python-pytest_1-Empty_free/2022-1c-recuperatorio-parcial1/20260921-203929-73794/workdir/src/customerimporter/dni.py
from customerimporter.customer_identification import CustomerIdentification


class Dni(CustomerIdentification):

    # type

    @classmethod
    def type_code(cls):
        return "D"

    # identification

    def dni_number_if_none(self, a_none_block):
        return int(self.number())

    def is_dni(self):
        return True

    # error messages

    @classmethod
    def invalid_value_error_description(cls):
        return "Invalid DNI number"

    # validation

    @classmethod
    def assert_is_valid(cls, an_identification_number):
        cls._assert_that(cls._are_all_digits(an_identification_number))
        cls._assert_that(int(an_identification_number) >= 1 and int(an_identification_number) <= 99999999)
